builder_utils/utils.go

```diff
@@ -558,5 +558,10 @@ func GetCachedCoreArchiveFileName(fqbn, coreFolder string) string {
 		coreFolder = absCoreFolder
 	} // silently continue if absolute path can't be detected
 	hash := utils.MD5Sum([]byte(coreFolder))
-	return "core_" + fqbnToUnderscore + "_" + hash + ".a"
+	realName := "core_" + fqbnToUnderscore + "_" + hash + ".a"
+	if len(realName) > 100 {
+		// avoid really long names, simply hash the final part
+		realName = "core_" + utils.MD5Sum([]byte(fqbnToUnderscore+"_"+hash)) + ".a"
```

Comment:
This feels a bit arbitrary, but it would probably work well in practice.

Reply:
Really arbitrary indeed; it still fails if someone has a TEMP directory with ~140 characters (`260 - 100 - len("arduino-core-cache-xxxxxx")`) but on Windows it would mean that your username is at least 100 keystrokes long (very unlikely)